src/Controls/src/Core/VisualElementProfiler/VisualElementProfiler.cs

```diff
@@ -0,0 +1,280 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Microsoft.Maui.Controls;
+
+internal class VisualElementProfiler
+{
+	class LayoutPassTypeStats : ILayoutPassTypeStats
+	{
+		public LayoutPassTypeStats(LayoutPassType layoutPassType, long count = 0, long standaloneCount = 0, long totalTime = 0)
+		{
+			LayoutPassType = layoutPassType;
+			Count = count;
+			StandaloneCount = standaloneCount;
+			TotalTime = totalTime;
+		}
+
+		public LayoutPassType LayoutPassType { get; }
+		public long StandaloneCount { get; set; }
+		public long Count { get; set; }
+		public long TotalTime { get; set; }
+	}
+
+	class LayoutPassStats : ILayoutPassStats
+	{
+		private readonly LayoutPassTypeStats[] _layoutPassTypeStats =
+		[
+			new(LayoutPassType.Measure),
+			new(LayoutPassType.CrossPlatformMeasure),
+			new(LayoutPassType.Arrange),
+			new(LayoutPassType.CrossPlatformArrange)
+		];
+
+		public LayoutPassTypeStats this[LayoutPassType type]
+		{
+			get
+			{
+				return _layoutPassTypeStats[(int)type];
+			}
+		}
+
+		ILayoutPassTypeStats ILayoutPassStats.this[LayoutPassType type]
+		{
+			get
+			{
+				return this[type];
+			}
+		}
+
+		public override string ToString()
+		{
+			var sb = new StringBuilder();
+			foreach (var typeStats in _layoutPassTypeStats)
+			{
+				sb.AppendLine($"{typeStats.LayoutPassType}: {typeStats.Count} ({typeStats.TotalTime} ns)");
+			}
+			return sb.ToString();
+		}
+	}
+
+	class LayoutPassTimer
+	{
+		// [ Measure, CrossPlatformMeasure, Arrange, CrossPlatformArrange ]
+		private readonly long[] _startTimes = [0, 0, 0, 0];
```

Comment:
Initializing just using `new long[4]` will automatically initializes all values to 0.